lib/trento/application/process_managers/deregistration_process_manager.ex

```diff
@@ -84,10 +144,56 @@ defmodule Trento.DeregistrationProcessManager do
     ]
   end
 
-  def apply(%DeregistrationProcessManager{} = state, %ClusterRolledUp{
-        cluster_id: cluster_id
-      }) do
-    %DeregistrationProcessManager{state | cluster_id: cluster_id}
+  def handle(
+        %DeregistrationProcessManager{
+          cluster_id: cluster_id,
+          database_instances: database_instances,
+          application_instances: application_instances
+        },
+        %HostDeregistrationRequested{
+          host_id: host_id,
+          requested_at: requested_at
+        }
+      ) do
```

Comment:
What about:
```
List.flatten([
    Enum.map(database_instances, fn %Instance{
    ...
    end),
    Enum.map(application_instances, fn %Instance{
    ...
    end),
    if cluster_id do
        %DeregisterClusterHost{...
    end,
    %DeregisterHost{...
])
```
This way we respect the logical order of commands as well
The `if` condition smells a bit and maybe there is a better way of doing this with pattern matching.